stonesoup/kernel.py

```diff
@@ -47,113 +57,228 @@ def _get_state_vectors(state1, state2):
+    def __call__(self, state1, state2=None, **kwargs):
+        r"""Calculate the Gaussian Kernel transformation for a pair of :class:`~.State` objects.
 
         Parameters
         ----------
-        state1 : :class:`~.KernelParticleState`
-        state2 : :class:`~.KernelParticleState`
+        state1 : :class:`~.State`
+        state2 : :class:`~.State`
 
         Returns
         -------
         StateVectors
             Transformed state vector in kernel space.
         """
+        self.update_parameters(kwargs)
         state_vector1, state_vector2 = self._get_state_vectors(state1, state2)
-        diff_tilde_x = (state_vector1.T[:, :, None] - state_vector2.T[:, None, :]) ** 2
+
+        diff_tilde_x = (state_vector1[:, :, None] - state_vector2[:, None, :]) ** 2
         diff_tilde_x_sum = np.sum(diff_tilde_x, axis=0)
 
         k_tilde_x = np.exp(-diff_tilde_x_sum/(2*self.variance)) / np.sqrt(2*np.pi*self.variance)
 
         return StateVectors(k_tilde_x)
+
+
+class TrackKernel(Kernel):
+    kernel: Kernel = Property(doc="Base Kernel class")
+    mapping: list = Property(default=None, doc="List of mappings of the components to be used in "
+                                               "the kernel from the state vector.")
+
+    @property
+    def parameters(self):
+        return self.kernel.parameters
+
+    def update_parameters(self, kwargs):
+        return self.kernel.update_parameters(kwargs)
+
+    def _get_states(self, state1, state2):
+        if state2 is None:
+            state2 = state1
+        if self.mapping is None:
+            self.mapping = list(range(state1[0].state_vector.shape[0]))
+        state1 = np.hstack([state.state_vector[self.mapping] for state in state1])
+        state2 = np.hstack([state.state_vector[self.mapping] for state in state2])
+        return state1, state2
+
+    def __call__(self, state1, state2=None, **kwargs):
+        r"""
+        Compute the kernel state of a pair of :class:`~.Track` objects
+
+        Parameters
+        ----------
+        state1 : :class:`~.Track`
+        state2 : :class:`~.Track`
+
+        Returns
+        -------
+        StateVectors
+            kernel state of a pair of input :class:`~.State` objects
+        """
+        state1, state2 = self._get_states(state1, state2)
+        return self.kernel.__call__(state1, state2, **kwargs)
+
+
+class MeasurementKernel(TrackKernel):
+
+    def __call__(self, state1, state2=None, **kwargs):
+        r"""
+        Compute the kernel state of a pair of lists of measurements as :class:`~.List` objects
+
+        Parameters
+        ----------
+        state1 : :class:`~.List`
+        state2 : :class:`~.List`
+
+        Returns
+        -------
+        StateVectors
+            kernel state of a pair of input :class:`~.State` objects
+        """
+        state1, state2 = self._get_states(state1, state2)
+        return self.kernel.__call__(state1, state2, **kwargs)
```

Comment:
These are both very similar. Maybe a `StateKernel` or `_StateKernel` that these inherit from?